lib/koans/19_genservers.ex

```diff
@@ -0,0 +1,141 @@
+      {:reply, "CRMO Series", current_state}
+    end
+
+    def handle_call(:get_owner_info, _from, current_state) do
+      {:reply, {:ok, "Argh...Jack Sparrow's password is: #{current_state}"}, current_state}
+    end
+
+    def handle_call(:get_multi, _from, current_state) do
+      {:reply, {:ok, ["this", "is", "sparta"]}, current_state}
+    end
+
+    def handle_call(:name_check, _from, current_state) do
+      {:reply, "Congrats! Your process was successfully named.", current_state}
+    end
+    
+    def handle_call({:unlock, password}, _from, current_password) do
+      case password do
+        password when password === current_password ->
+          {:reply, {:ok, "Bicycle unlocked!"}, current_password}
+        _ -> 
+          {:reply, {:error, "Incorrect password!"}, current_password}
+      end
+    end
+    
+    def handle_cast({:reset, old_password, new_password}, current_password) do
+      case old_password do
+        old_password when old_password == current_password ->
+          {:noreply, new_password}
+        _ ->
+          {:noreply, current_password} 
+      end
+    end
+  end
+
+  koan "Servers that are created and initialized successfully returns a tuple that holds the PID of the server" do
+    {:ok, pid} = GenServer.start_link(BicycleLock, nil)
+    assert is_pid(pid) == ___
+  end
+
+  koan "When starting a GenServer you can set it's initial state" do
+    {:ok, pid} = GenServer.start_link(BicycleLock, "Hey Arnold!")
+    assert GenServer.call(pid, :get_password) == ___
+  end
+
+  koan "The handle_call callback is synchronous so it will block until a reply is received" do
+    {:ok, pid} = GenServer.start_link(BicycleLock, nil)
+    assert GenServer.call(pid, :get_password) == ___
+  end
+
+  koan "A server can support multiple actions by implementing multiple handle_call functions" do
+    {:ok, pid} = GenServer.start_link(BicycleLock, nil)
+    assert GenServer.call(pid, :get_bike_brand) == ___
+    assert GenServer.call(pid, :get_bike_name) == ___
+  end
+
+  koan "A handler can return multiple values" do
+    {:ok, pid} = GenServer.start_link(BicycleLock, nil)
+    {:ok, multi_values} = GenServer.call(pid, :get_multi)
+    assert multi_values == ___
+  end
+
+  koan "The handle_cast callback handles asynchronous messages" do
+    {:ok, pid} = GenServer.start_link(BicycleLock, nil)
+    GenServer.cast(pid, {:reset, 1234, "Hello"})
+    assert GenServer.call(pid, :get_password) == ___
+  end
+
+  koan "Handlers can also return error responses" do
+    {:ok, pid} = GenServer.start_link(BicycleLock, nil)
+    assert GenServer.call(pid, {:unlock, 2017}) == ___
+  end
+
+  koan "Referencing processes by their PID gets old pretty quickly, so let's name them" do
+    {:ok, pid} = GenServer.start_link(BicycleLock, nil, name: :bike_lock)
+    assert GenServer.call(:bike_lock, :name_check) == ___
+  end
+
+  koan "Our server works but it's pretty ugly to use; so lets use a cleaner interface" do
```

Comment:
It might be clearer to say something like "It's tiring to repeat GenServer.start_link. A common pattern is adding a function to handle this."